node/impl/full.go

```diff
@@ -117,4 +123,82 @@ func (n *FullNodeAPI) NodeStatus(ctx context.Context, inclChainStatus bool) (sta
 	return status, nil
 }
 
+// This method had been kept separate from wallet group implementation as it is not required for basic lotus functions
+// This method is exclusively used for Non Interactive Authentication based on worker address
+func (n *FullNodeAPI) FilIdSp(ctx context.Context, maddr address.Address) ([]byte, error) {
+	head, err := n.ChainHead(ctx)
+	if err != nil {
+		return nil, xerrors.Errorf("failed to get chain head: %w", err)
+	}
+
+	finality := abi.ChainEpoch(uint64(head.Height()) - 900)
+	ftipset, err := n.ChainGetTipSetByHeight(ctx, finality, types.EmptyTSK)
+	if err != nil {
+		return nil, xerrors.Errorf("failed to get finality tipset: %w", err)
+	}
+	beacon, err := n.StateGetBeaconEntry(ctx, head.Height())
+	if err != nil {
+		return nil, xerrors.Errorf("failed to get DRAND beacon: %w", err)
+	}
+
+	var msg []byte
+
+	// Append prefix to avoid making a viable message
+	msg = append(msg, []byte("   ")...)
+	// Append beacon
+	msg = append(msg, beacon.Data...)
+
+	// Get miner info for last finality
+	minfo, err := n.StateMinerInfo(ctx, maddr, ftipset.Key())
+	if err != nil {
+		return nil, xerrors.Errorf("failed to get miner info: %w", err)
+	}
+
+	worker := minfo.Worker
+	var sig *crypto.Signature
+
+	sig, err = n.WalletSign(ctx, worker, msg)
+	if err != nil {
+		return nil, xerrors.Errorf("failed to get sign message with worker address key: %w", err)
+	}
+
+	return append([]byte{byte(sig.Type)}, sig.Data...), nil
```

Comment:
This is just the sig, you need to output a complete AUTH line, looks like:
```
FIL-SPID-V0 2219681;f016309;2;oaNMQkK8rwPriM8Os6N428tJJSdLvcQJAT/jhY3E5D/3NkXlmguBfEonsK1cH4VnB1NaCgz0iWXJt0F/HideSHWQ9MwPW6RSFJebuvKMaGO8z05QNk4Cipaui9vrHiY1
```

Reply:
@LexLuthr now that I look at this and think it through: my adding the `;2;` was a mistake and has to go: it is always clear what created the sig:
in the case of SPID it is a BLS
in the case of ADDRID it will be the pubkey itself:
`FIL-ADDRID-V0 2219681;f1xxxxxxxxxxxxx;somesiggggggggg`